pandas/tests/io/parser/test_common.py

```diff
@@ -2079,3 +2080,17 @@ def test_integer_precision(all_parsers):
     result = parser.read_csv(StringIO(s), header=None)[4]
     expected = Series([4321583677327450765, 4321113141090630389], name=4)
     tm.assert_series_equal(result, expected)
+
+
+@td.skip_if_no("psutil")
+def test_file_descriptor_leak(all_parsers):
```

Comment:
We already have some machinery to wrap the function and check for a leak here:

https://github.com/pandas-dev/pandas/blob/25e2f8687f0a7033ddbbd6aa1b833d0586f99bad/pandas/util/_test_decorators.py#L236

Can you use this instead?

Reply:
I tried it, but it doesn't trigger on this specific issue. Apparently the file handle gets destroyed by some other means before the check can trigger.
But I can double check.

Reply:
I can choose between using the `tm.ensure_clean` context manager which will close the file as well and therefore prevent the file descriptor leak, or use the `check_file_leaks` context manager, but not both.
Please express a preference of one above the other.

Reply:
I think should use the `check_file_leaks` context manager

Reply:
OK, so that means I have to ditch the `tm.ensure_clean` call?

Reply:
I'm not clear why they can't both be used together - what error were you getting?

Reply:
Well, the `tm.ensure_clean` basically resolves the file descriptor leaked by `read_csv()`, so that the enclosing `check_file_leaks` does not trigger.
I always make sure that the test fails when I undo my change. And with both in place, the test succeeds even when the bug is present. Therefore the test becomes useless when both constructs are used in combination.

Reply:
How are you setting that up? I see this as raising on master:

```python
with tm.ensure_clean() as path:
    def shouldnt_leak():
        try:
            pd.read_csv(path)
        except EmptyDataError:
            pass
        
    check_file_leaks(shouldnt_leak)()
```

Though admittedly intermittently

Reply:
That could work, because now the file leak checking is inside the `tm.ensure_clean` context manager.
I used it as a decorator of the test in this fashion:
```python
@td.check_file_leaks
def test_bla():
    with tm.ensure_clean() as path:
        try:
            pd.read_csv(path)
        except EmptyDataError:
            pass
```

Reply:
This seems to work for me, so it is now changed to use td.check_file_leaks in the right manner.